pyClient/zeth/joinsplit.py

```diff
@@ -601,46 +513,55 @@ def _compute_proof_hashes(proof_json: GenericProof) -> Tuple[bytes, bytes]:
         if key != "inputs":
             proof_elements.extend(proof_json[key])
     return (
-        sha256(encode_message_to_bytes(proof_elements)).digest(),
-        sha256(encode_pub_input_to_hash(proof_json["inputs"])).digest())
+        encode_message_to_bytes(proof_elements),
+        encode_message_to_bytes(proof_json["inputs"]))
 
 
 def sign_mix_tx(
+        signing_keypair: SigningKeyPair,  # Ephemeral signing key, tied to proof
```

Comment:
Should be a JoinsplitSigKeyPair, which itself is a wrapper around `SigningKeyPair`

```python
# JoinSplit Signature Keys definitions
JoinsplitSigKeyPair = signing.SchnorrKeyPair
JoinsplitSigPrivKey = signing.SchnorrSigningKey
JoinsplitSigPubKey = signing.SchnorrVerificationKey
```

Also, let's rename `sign_mix_tx` to `sign_joinsplit`, because we do not sign the transaction per say (ECDSA signed), here we only sign the tx_data corresponding to the Joinsplit description, so let's make it clear.

 (see: https://github.com/clearmatics/zeth/pull/122)

Reply:
renamed

Reply:
Added the type wrappers back, renamed the sign functtion.